### ISSUE TITLE
Failed to load the extension, No module named 'zarr'

### ISSUE THREAD

**Issue by GrishDaProgrammer** [wawawario2/long_term_memory#40]
Hello. Fresh build of oobabooga on windows, and a fresh build of the long_term_memory extension is yielding the error lower in this post. (No module named zarr).

Relevant information:
Windows 10 OS
Zarr exists in the directory: C:\Users\XX\AppData\Local\Programs\Python\Python310\Lib\site-packages\zarr
Zarr exists on PATH / System variables.
Step 3 in the README (pytest for extension, passes all tests)

2023-07-04 22:03:49 INFO:Loading the extension "long_term_memory"...
2023-07-04 22:03:49 ERROR:Failed to load the extension "long_term_memory".
Traceback (most recent call last):
  File "C:\oobabooga_2\oobabooga_windows\oobabooga_windows\text-generation-webui\modules\extensions.py", line 36, in load_extensions
    exec(f"import extensions.{name}.script")
  File "<string>", line 1, in <module>
  File "C:\oobabooga_2\oobabooga_windows\oobabooga_windows\text-generation-webui\extensions\long_term_memory\script.py", line 14, in <module>
    from extensions.long_term_memory.core.memory_database import LtmDatabase
  File "C:\oobabooga_2\oobabooga_windows\oobabooga_windows\text-generation-webui\extensions\long_term_memory\core\memory_database.py", line 10, in <module>
    import zarr
ModuleNotFoundError: No module named 'zarr'
2023-07-04 22:03:49 INFO:Loading the extension "gallery"...
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.

I am not the strongest at open source coding/python, so I would appreciate any help I can get with this issue.

**Comment by luthis1124** [comment]
while in the virtual env, can you pip install zarr?

**Comment by GrishDaProgrammer** [comment]
I was able to fix this on my own, and I have been meaning to post the solution here.

For reference, pip install zarr would have yielded either a "it's already installed" or a successful reinstall.

The problem was a dependency that isn't satisfied automatically on windows 10. It has to do with Microsoft C++ build tools missing. If you like history: https://github.com/oobabooga/one-click-installers/issues/56

The fix:
1.) Download build tools; https://visualstudio.microsoft.com/visual-cpp-build-tools/
2.) Follow this russian(?) image from the above one-click-installer linked page, and make sure that package is installed:
![image](https://github.com/wawawario2/long_term_memory/assets/116243185/9020a78f-9a3f-4ef5-bf04-80d9f4fadf68)
3.) I restarted my pc. Not sure if required.
4.) Run update.bat 

This worked for me atleast. Hopefully someone else can confirm.
